libs/zk/include/nil/crypto3/zk/commitments/polynomial/lpc.hpp

```diff
@@ -421,6 +523,60 @@ namespace nil {
                         eval_storage_type z;
                         typename basic_fri::proof_type fri_proof;
                     };
+
+                    // Represents an initial proof, which must be created for each of the N provers.
+                    struct initial_proof_type {
+                        bool operator==(const initial_proof_type &rhs) const {
+                            return initial_fri_proof == rhs.initial_fri_proof && z == rhs.z;
+                        }
+
+                        bool operator!=(const initial_proof_type &rhs) const {
+                            return !(rhs == *this);
+                        }
+
+                        eval_storage_type z;
+                        typename basic_fri::initial_proofs_batch_type initial_fri_proof;
+                    };
+
+                    // Represents a round proof, which must be created just once on the main prover.
+                    struct round_proof_type {
```

Comment:
`fri_proof_type` maybe?

We use round_proof term twice. Once it is a thing that is done by the main prover, and second time is a part of round_proofs_batch_type

It will be easy fri_proof contains 
*  `fri_round_proof(s)` 
*  `fri_commitments_proof_part`